src/PrestaShopBundle/Form/Admin/Type/SubmittableDeltaQuantityType.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\CallbackTransformer;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Type;
+
+/**
+ * Wraps SubmittableInput and DeltaQuantity components to work together.
+ * admin-dev/themes/new-theme/js/components/form/submittable-delta-quantity-input.ts responsible for javascript part.
+ *
+ * @see DeltaQuantityType
+ * @see SubmittableInputType
+ */
+class SubmittableDeltaQuantityType extends DeltaQuantityType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        $builder->add('delta', SubmittableInputType::class, [
```

Comment:
On second though I don't think we should use a `SubmittableInputType` here because it adds an extra layer `value` in the form field and data

Instead we could use a regular `NumberType` and mimic the behaviour of the submittable input in the twig UI kit (which you already do thanks to the `submittable_delta_quantity_delta` block prefix) Maybe you're gonna have to copy the code in both widgets but at least the form structure of this component will be cleaner

Although the duplication may be avoided with a common block relying on a variable `input_form` that could be set in each widget blocks like:

```twig
{% block submittable_delta_quantity_delta_widget %}
  {%- set attr = attr|merge({class: (attr.class|default('') ~ ' delta-quantity-delta')|trim}) -%}
  {% set input_form = form %}
  {{- block('submittable_input_widget') -}}
{% endblock %}

{% block submittable_input_widget %}
  {%- set attr = attr|merge({'class': (attr.class|default('') ~ ' ps-submittable-input-wrapper')|trim}) -%}
  {% set input_form = input_form|default(form.value) %}
  <div {{ block('widget_attributes') }}>
    {%- set typeAttr = form.vars.type_attr|merge({
      'class': (form.vars.type_attr.class|default('') ~ ' submittable-input')|trim,
      'data-initial-value': form.value.vars.value,
      'value': form.value.vars.value,
    }) -%}

    {{ form_widget(input_form, {'attr': typeAttr}) }}
    <button type="button" class="check-button d-none">
      <i class="material-icons">check</i>
    </button>
  </div>
{% endblock submittable_input_widget %}
```

(not it sure this would work actually 😅 )

Reply:
can't pass variable between blocks.
Made it work by duplicating some things.